person.go

```diff
@@ -102,71 +102,81 @@ var (
 )
 
 type Person struct {
-	Faker *Faker
+	Faker  *Faker
+	gender string
```

Comment:
I guess is better to have a `func (p Person) Gender() string` instead of adding new fields here, but I like the idea of creating structs with filled data, maybe we can have another PR with a method for that?

Reply:
Btw, if I just create `func (p Person) Gender() string`, we don't know the name will be generated from `func Name()` (male/female). It's just generate a randomly. This is also my reason creating gender inside the struct field, to catch up the name generated from func Name(). Is it make sense?

If we make gender without related to the name, yes, we just create `func Gender() string` directly.

Reply:
actually what you can is this:

for `NameMale()` create a function that concats `FirstNameMale()` and `LastName()`, for `NameFemale()` do the same but with `FirstNameFemale()` and `LastName()`.

did you get it?

Reply:
OK, i'll do that. I just mind if we create `NameMale()` that you mentioned, it's just creating a random male name with specified gender, not creating a random name (female/male) and that come with her/his gender.

Sorry for this long discussion. And thank you for your advices 🙏

Reply:
no problem, I really appreciate your effort to do the best here, maybe I have misunderstood something, you want to generate a name and gender at the same time, I just want to avoid changing the current behavior of the `Name()` method and avoid using an internal field, what could be done then is create a `GenderMale()` that returns the word `male`, and a `GenderFemale()` that returns the word `female`, plus the usage of those methods in the `Gender()` method (putting those methods instead of the hard-coded values).

So for your point what we can do then with those methods is this:
```
genderMale := faker.Person().GenderMale()
nameMale := faker.Person().NameMale()
```

do you think this could cover your point?

Reply:
My points is (with some examples)

```
name, gender := faker.Person().Name() // result ("John Lennon", "Male")
name, gender := faker.Person().Name() // result ("Ringo Starr", "Male")
name, gender := faker.Person().Name() // result ("Rihanna", "Female")
name, gender := faker.Person().Name() // result ("Mick Jagger", "Male")
name, gender := faker.Person().Name() // result ("Beyonce", "Female")
```

It's like, we generating randomly name whatever the gender, but the return still knows that name is male/female. But the point of you is, we specified the gender like `nameMale()` or `nameFemale()` and then return the random name with specified gender.

Sorry, maybe I explaining in a bad way 😭

Reply:
no worries, I guess now I understood, in this case we can have another method `NameAndGender()` that will randomly return based of a boolean, see this code example here https://play.golang.org/p/TFnX8W9yZSp

Reply:
Yes! that cover the use case. it even turned out your solution is more simple than mine haha.

If that use case is ok, I'll be updating the code. But, before doing that, I make new PR for `bool` job that related to this use case.

Reply:
yep, I reviewed, I have some minor suggestions there, feel free to check/question any of my comments there